apps/mobile/src/features/browser/approver-sheet/stx/transfer-sip10-ft/transfer-sip10-ft.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from 'react';
+
+import { useToastContext } from '@/components/toast/toast-context';
+import { BaseStxTxApproverLayout } from '@/features/approver/layouts/base-stx-tx-approver.layout';
+import { getTxOptions } from '@/features/approver/utils';
+import { useBroadcastStxTransaction } from '@/queries/stacks/use-broadcast-stx-transaction';
+import { useAccounts } from '@/store/accounts/accounts.read';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { assertStacksSigner } from '@/store/keychains/stacks/utils';
+import { useNetworkPreferenceStacksNetwork } from '@/store/settings/settings.read';
+import { deserializeTransaction } from '@stacks/transactions';
+
+import {
+  RpcRequest,
+  RpcResponse,
+  createRpcSuccessResponse,
+  stxTransferSip10Ft,
+} from '@leather.io/rpc';
+
+import { useTransferSip10FtTxHex } from './hooks';
+
+interface TransferSip10FtApproverProps {
+  request: RpcRequest<typeof stxTransferSip10Ft>;
+  sendResult(result: RpcResponse<typeof stxTransferSip10Ft>): void;
+  closeApprover(): void;
+  nonce: number;
+  accountId: string;
+}
+
+export function TransferSip10FtApprover({
+  request,
+  closeApprover,
+  sendResult,
+  nonce,
+  accountId,
+}: TransferSip10FtApproverProps) {
+  const { displayToast } = useToastContext();
+  const network = useNetworkPreferenceStacksNetwork();
+
+  const { list: accounts } = useAccounts();
+  const [txHex, setTxHex] = useState<null | string>(null);
+  useTransferSip10FtTxHex({ request, accountId, setTxHex, nonce });
```

Comment:
I got a bit confused here as we don't use any values from `useTransferSip10FtTxHex`. 

I'm curious if it needs to be a hook? 

Maybe we could instead pass in `fromAccountId` also? 

**  const { fromAccountId } = useStacksSigners();**